File: broker/src/main/java/lambdamart/service/broker/graphql/Query.java
package lambdamart.service.broker.graphql;

import java.util.List;

import org.springframework.stereotype.Component;

import lambdamart.service.broker.graphql.model.Vendor;


@Component
public class Query implements GraphQLQueryResolver {

    // Add methods to handle 'vendors', 'product', 'products', and 'item' queries

    // Example:
    public List<Vendor> vendors() {
        // Call methods to fetch vendors from MongoDB or vendor GraphQL clients
    }

    // Similarly, implement other query methods
}

File: ts-broker/service/src/schema/query.ts
import { gql } from "graphql-request";

export const typeDef = gql`
  # Query the broker for information about vendors and products
  type Query {
    vendors(vendorIds: [ID!]): [Vendor!]!
    product(classId: ID!): Product
    products(
      coarseClassNames: [String!]
      productTypes: [String!]
      classIds: [ID!]
    ): [Product!]!
    item(vendorIds: [ID!], itemId: ID!): [InventoryItem!]
  }
`;
